gnoblog-cli/cmd/post.go

```diff
@@ -0,0 +1,261 @@
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/gnolang/gno/gno.land/pkg/gnoclient"
+	"github.com/gnolang/gno/gnovm/pkg/gnoenv"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"github.com/peterbourgon/ff/v3/ffcli"
+	"os"
+	"strings"
+	"time"
+)
+
+type cliCfg struct {
+	Publish       bool
+	Edit          bool
+	GasWanted     int64
+	GasFee        string
+	ChainId       string
+	BlogRealmPath string
+
+	KeyName               string
+	GnoHome               string
+	Remote                string
+	Quiet                 bool
+	InsecurePasswordStdIn bool
+}
+
+func newPostCommand() *ffcli.Command {
+	var (
+		fs  = flag.NewFlagSet("post", flag.ExitOnError)
+		cfg = &cliCfg{}
+	)
+
+	// Register flagset
+	cfg.registerFlags(fs)
+
+	// Make Post command
+	return &ffcli.Command{
+		Name:       "post",
+		ShortUsage: "post <FILE OR FILES_DIR> [flags]",
+		LongHelp:   `Post one or more files. Passing in a file will post that single file, while passing in a directory will search for all .md files and batch post them.`,
+		FlagSet:    fs,
+		Exec:       cfg.execPost,
+	}
+}
+
+// Register flags
+func (cfg *cliCfg) registerFlags(fs *flag.FlagSet) {
+	fs.StringVar(&cfg.KeyName,
+		"key",
+		"",
+		"name of keypair to use for deployment",
+	)
+	fs.BoolVar(&cfg.Publish,
+		"publish",
+		false,
+		"publish blogpost",
+	)
+	fs.BoolVar(&cfg.Edit,
+		"edit",
+		false,
+		"edit mode",
+	)
+	fs.Int64Var(&cfg.GasWanted,
+		"gas-wanted",
+		5000000,
+		"gas requested for tx",
+	)
+	fs.StringVar(&cfg.GasFee,
+		"gas-fee",
+		"1000000ugnot",
+		"gas payment fee",
+	)
+	fs.StringVar(&cfg.ChainId,
+		"chainid",
+		"dev",
+		"chain ID",
+	)
+	fs.StringVar(&cfg.BlogRealmPath,
+		"pkgpath",
+		"gno.land/r/gnoland/blog",
+		"blog realm path",
+	)
+	fs.StringVar(&cfg.GnoHome,
+		"home",
+		gnoenv.HomeDir(),
+		"home directory",
+	)
+	fs.StringVar(&cfg.Remote,
+		"remote",
+		"localhost:26657",
+		"remote node URL",
+	)
+	fs.BoolVar(&cfg.InsecurePasswordStdIn,
+		"insecure-password-stdin",
+		false,
+		"WARNING! take password from stdin",
+	)
+}
+
+func (cfg *cliCfg) execPost(_ context.Context, args []string) error {
+	if len(args) > 1 {
+		fmt.Println(args)
+
+		return errInvalidNumberOfArgs
+	}
+
+	fileInfo, err := os.Stat(args[0])
+	if err != nil {
+		return errInvalidPath
+	}
+
+	// Init IO for password input
+	io := commands.NewDefaultIO()
+
+	var pass string
+	if cfg.Quiet {
+		pass, err = io.GetPassword("", cfg.InsecurePasswordStdIn)
+	} else {
+		pass, err = io.GetPassword("Enter password:", cfg.InsecurePasswordStdIn)
+	}
+	if err != nil {
+		return err
+	}
+
+	// Initialize Gnoclient
+	client := gnoclient.Client{
+		Signer:    initSigner(cfg, pass),
+		RPCClient: initRPCClient(cfg),
+	}
+
+	// Batch post request passed in
+	if fileInfo.IsDir() {
+		return cfg.batchPost(client, args[0])
+	}
+
+	// Single post request passed in an argument
+	return cfg.singlePost(client, args[0])
+}
+
+func (cfg *cliCfg) singlePost(c gnoclient.Client, postPath string) error {
```

Comment:
Please remove the use of `singlePost` and instead utilize `post`, which will support both batches and single posts.

Reply:
I unified the `singlePost` & `batchPost` functions: [68778e0](https://github.com/gnolang/blog/pull/63/commits/68778e00208f3e8385f5785a3a2bbbaa17794b1a)